IntegrationWithPharmacies/Controllers/SharingPrescriptionController.cs

```diff
@@ -1,30 +1,198 @@
+            StreamReader readStream = new StreamReader(response, System.Text.Encoding.GetEncoding("utf-8"));
+            string description = readStream.ReadToEnd();
+            if (description.Length != 0) return Ok(description);
+            return BadRequest();
+            
+        }
+        [HttpGet("http/medicineAvailability/{medicine}")]
+        public IActionResult GetMedicineAvailability(String medicine)
+        {
+            HttpWebRequest webRequest = (HttpWebRequest)WebRequest.Create("http://localhost:8082/medicinePharmacy/" + medicine);
+            HttpWebResponse webResponse = (HttpWebResponse)webRequest.GetResponse();
+            Stream response = webResponse.GetResponseStream();
+            StreamReader readStream = new StreamReader(response, System.Text.Encoding.GetEncoding("utf-8"));
+            return FormMedicineAvailability(readStream.ReadToEnd());
+        }
+
+        private IActionResult FormMedicineAvailability(string availability)
+        {
+            if (availability.Length > 5) return GetMedicineAvailabilityTable(availability);
+            return BadRequest();    
+        }
+
+        private IActionResult GetMedicineAvailabilityTable(string availability)
+        {
+            List<MedicineName> medicines = new List<MedicineName>();
+            String[] fileParts = availability.Split(";");
+            if (fileParts.Length == 0) GetOnlyOnePharmacy(availability, medicines);
+            else GetAllPharmacies(medicines, fileParts);
+            return Ok(medicines);
+        }
+
+        private static void GetAllPharmacies(List<MedicineName> medicines, string[] fileParts)
+        {
+            for (int i = 0; i < fileParts.Length; i++)
+            {
+                String[] nameParts = fileParts[i].Split("_");
+                medicines.Add(new MedicineName("Pharmacy: " + nameParts[0] + ", city: " + nameParts[1], nameParts[2]));
+            }
+        }
+
+        private static void GetOnlyOnePharmacy(string availability, List<MedicineName> medicines)
+        {
+            String[] nameParts = availability.Split("_");
+            MedicineName medicine = new MedicineName("Pharmacy: " + nameParts[0] + ", city: " + nameParts[1], nameParts[2]);
+            medicines.Add(medicine);
+        }
+
+        private int getRandomNumber()
+        {
+            return new Random().Next(1, 100);
+        }
+        private String getTextForPrescription(Prescription prescription)
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            return stringBuilder.Append(prescription.Pharmacy+"          Precription for medicine\n\nPatients name: " + prescription.Name + "\nPatients surname: " + prescription.Surname + "\nPatients medical ID number: " + prescription.MedicalIDNumber + "\nMedication: " + prescription.Medicine + "     Quantity: " + prescription.Quantity + "\nUsage: " + prescription.Usage + "\n").ToString();

```

Comment:
Write this in form:
"return prescription.Pharmacy+"          Precription for medicine\n\nPatients name: " + prescription.Name + "\nPatients surname: " + prescription.Surname + "\nPatients medical ID number: " + prescription.MedicalIDNumber + "\nMedication: " + prescription.Medicine + "     Quantity: " + prescription.Quantity + "\nUsage: " + prescription.Usage + "\n""